applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp

```diff
@@ -583,6 +588,70 @@ class EigensolverStrategy
     ///@name Private Operations
     ///@{
 
+    /**
+     * @brief Recover the solution related to the slave dofs in case of master-slave constraints.
+     * @details This function is an adaptation of the implementation in ConstraintUtilities,
+     *  since there the variables are assumed to be stored in SolutionStepValue.
+     *  Beware that this implementation is only valid for Block B&S, since the master-slave constraints
+     *  don't work with Elimination B&S yet.
+     */
+    void ReconstructSolution(
+        DenseMatrixType& rEigenvectors
+    )
+    {
+        KRATOS_TRY
+
+        auto& r_model_part = BaseType::GetModelPart();
+        const std::size_t number_of_eigenvalues = rEigenvectors.size1();
+
+        struct TLS{
+            Matrix relation_matrix;
+            Vector constant_vector;
+            Vector master_dofs_values;
+        };
+
+        for (std::size_t i_eigenvalue = 0; i_eigenvalue < number_of_eigenvalues; ++i_eigenvalue){
+
+            // Reset slave dofs
+            block_for_each(r_model_part.MasterSlaveConstraints(), [&i_eigenvalue, &rEigenvectors](const MasterSlaveConstraint& r_master_slave_constraint){
+                const auto& r_slave_dofs_vector = r_master_slave_constraint.GetSlaveDofsVector();
+                for (const auto& r_slave_dof: r_slave_dofs_vector){
+                    rEigenvectors(i_eigenvalue, r_slave_dof->EquationId()) = 0.0;
```

Comment:
the slave dof can exist several time no?
In this case this would be a race condition
=> do the same as [here](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/constraints/linear_master_slave_constraint.cpp#L46) in this case

(AtomicMult will only be added in #8155)